packages/hardhat-plugin/src/ui/pretty-event-handler.ts

```diff
@@ -309,6 +312,77 @@ export class PrettyEventHandler implements ExecutionEventListener {
     };
   }
 
+  public ledgerConnectionStart(): void {
+    this.state = {
+      ...this.state,
+      ledger: true,
+      ledgerMessage: "Connecting wallet",
+    };
+
+    this._redisplayCurrentBatch();
+
+    this.state = {
+      ...this.state,
+      ledgerMessageIsDisplayed: true,
```

Comment:
Can you walk me through the need for the split here and have two state updates?
Why can't `ledgerMessageIsDisplayed: true` go in the first state update?